src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/8.diff.sql

```diff
@@ -0,0 +1,411 @@
+/*************************************************************
+    Migration 8: Adds Version and Fragment columns
+    to UriSearchParam to support Canonical searches
+**************************************************************/
+
+BEGIN TRANSACTION
+
+CREATE TYPE dbo.UriSearchParamTableType_2 AS TABLE
+(
+    SearchParamId smallint NOT NULL,
+    Uri varchar(256) COLLATE Latin1_General_100_CS_AS NOT NULL,
+    Version varchar(64) COLLATE Latin1_General_100_CS_AS NULL,
+    Fragment varchar(256) COLLATE Latin1_General_100_CS_AS NULL
+)
+
+GO
+
+ALTER TABLE dbo.UriSearchParam
+    ADD Version varchar(64) COLLATE Latin1_General_100_CS_AS NULL,
+        Fragment varchar(256) COLLATE Latin1_General_100_CS_AS NULL;
+
+GO
+
+CREATE NONCLUSTERED INDEX IX_UriSearchParam_SearchParamId_UriVersionFragment
+ON dbo.UriSearchParam
+(
+    SearchParamId,
+    Uri,
+    Version,
```

Comment:
Did you weigh included vs index columns?

Reply:
...I'm not sure if I understand the documentation correctly, but it seems like Included columns might be the way to go.
The query will always contain Uri, and optionally Version and Fragment